cmd/network/setup_linux.go

```diff
@@ -99,6 +95,85 @@ func main() {
 		logrus.Fatal(err)
 	}
 
+	connFile, err := vsockConn.File()
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	vmSwitchCmd := configureVMSwitch(ns,
+		vmSwitchLogFile,
+		vmSwitchPath,
+		tapIface,
+		subnet,
+		tapDeviceMacAddr,
+		connFile)
+
+	if err := vmSwitchCmd.Start(); err != nil {
+		logrus.Fatalf("could not start the vm-switch process: %v", err)
+	}
+	logrus.Infof("successfully started the vm-switch running with a PID: %v", vmSwitchCmd.Process.Pid)
+
+	if err := createVethPair(
+		defaultNamespacePID,
+		vmSwitchCmd.Process.Pid,
+		defaultNSVeth,
+		rancherDesktopNSVeth); err != nil {
+		logrus.Fatal(err)
+	}
+
+	if err := configureVethPair(rancherDesktopNSVeth, "192.168.1.2"); err != nil {
+		logrus.Fatalf("failed setting up veth: %s for rancher desktop namespace: %v", rancherDesktopNSVeth, err)
+	}
+
+	// switch back to the original namespace to configure veth0
```

Comment:
Hmm, switching _back_ to the original namespace has a kind of smell for me, but I can't say that it's _wrong_… I guess this just means we should be extra-careful that this is what we want.

It seems like we can probably use [`LinkSetNsFd`](https://pkg.go.dev/github.com/vishvananda/netlink#LinkSetNsFd) here instead?